### PR TITLE
Bug/id 461/Improve the main screen search engine

### LINKED ISSUE #138: Improve the user experience with the general search engine on the main screen.
**Describe the bug**

The bug occurs when the user searches for a specific location and the textField on the main screen is set and when the user clicks somewhere else on the screen the waypoint disappears but the browser still keeps the text.

**To Reproduce**

- Install the Hermes app.
- Click on the "Sign in with Google" button.
- Enter google credentials per existing account or per login.
- Select "Administrator" or "Genera" in the selector.
- Click the "Sign up" button.
- Depending on the selection you make you will be sent to the verification window or not.
- In the main window press the search button.
- Search for a place.
- Select the place you are looking for to mark the waypoint and the TextField in the main window.
- Click on another part of the map to perform another action.
- The waypoint disappears but the text in the TextField remains.

**Expected behavior**

The text in the search field should be cleared when the waypoint disappears.

**Smartphone (please complete the following information):**
 - Device: POCO X3 pro
 - OS: Android 12

**Screenshots**

Screen1

Screen2

### PR DESCRIPTION
**Pull request by Karvlox**
## Describe the bug

The bug occurs when the user searches for a specific location and the textField on the main screen is set and when the user clicks somewhere else on the screen the waypoint disappears but the browser still keeps the text.

## To Reproduce

Install the Hermes app.
Click on the "Sign in with Google" button.
Enter google credentials per existing account or per login.
Select "Administrator" or "Genera" in the selector.
Click the "Sign up" button.
Depending on the selection you make you will be sent to the verification window or not.
In the main window press the search button.
Search for a place.
Select the place you are looking for to mark the waypoint and the TextField in the main window.
Click on another part of the map to perform another action.
The waypoint disappears but the text in the TextField remains.

## Expected behavior

The text in the search field should be cleared when the waypoint disappears.

## Screenshots

Screen1

Screen2

## Smartphone 

Device: POCO X3 pro
OS: Android 12

### CODE REVIEW

**Review by VillalobosGarciaGaby — APPROVED** [review]

**Review by JeferssonCL — APPROVED** [review]
No comments

File: app/src/main/res/layout/activity_main.xml [hunk 1]
```diff
@@ -88,12 +88,23 @@
...
+                        android:textColorHint="@color/black"
+                        android:textSize="14sp" />
+
+                    <ImageButton
+                        android:id="@+id/buttonClear"
+                        android:layout_width="wrap_content"
+                        android:layout_height="20dp"
+                        android:layout_marginStart="245dp"
+                        android:layout_marginTop="-27dp"
+                        android:background="#000000"
```
Comment (JuanPabloArequipa) [inline app/src/main/res/layout/activity_main.xml]: Use the color black constant, it is at colors.xml

File: app/src/main/java/com/isc/hermes/MainActivity.java [hunk 2]
```diff
@@ -56,14 +56,15 @@ public class MainActivity extends AppCompatActivity implements OnMapReadyCallbac
...
-    private boolean visibilityMenu = false;
     private TextView searchView;
     private SharedSearcherPreferencesManager sharedSearcherPreferencesManager;
     private MarkerManager markerManager;
     private ActivityResultLauncher<Intent> launcher;
     private DrawerLayout drawerLayout;
     private NavigationView navigationView;
     private Toolbar toolbar;
+    private ImageButton buttonClear;
+    private String resetSearch = "Search...";
```
Comment (JuanPabloArequipa) [inline app/src/main/java/com/isc/hermes/MainActivity.java]: Watching the usage of this variable, a more appropiate name should be "searchText"

Comment (Karvlox) [reply app/src/main/java/com/isc/hermes/MainActivity.java]: Done.

File: app/src/main/res/layout/activity_main.xml (hunk 1 above)
Comment (Karvlox) [reply app/src/main/res/layout/activity_main.xml]: Done.

**Review by JuanPabloArequipa — APPROVED** [review]

**Pull request merged**